src/parser.ts

```diff
@@ -2583,12 +2583,34 @@ export class Parser {
     parseThrowStatement(): Node.ThrowStatement {
         const node = this.createNode();
         this.expectKeyword('throw');
+        const hasLineTerminator = this.hasLineTerminator;
+        const source = this.scanner.source;
 
-        if (this.hasLineTerminator) {
-            this.throwError(Messages.NewlineAfterThrow);
+        let argument;
+        try {
+            argument = this.parseExpression();
+        } catch (e) {
+            if (!hasLineTerminator) {
+                throw e;
+            }
+        }
+
+        if (hasLineTerminator) {
+            const isTemplateLiteral =
+                argument && argument.type === 'TemplateLiteral';
+            const hasLineTerminatorBeforeTemplate =
+                isTemplateLiteral &&
+                source.trim().indexOf('\n') < source.trim().indexOf('`');
+
+            if (
+                !argument ||
+                !isTemplateLiteral ||
+                hasLineTerminatorBeforeTemplate
+            ) {
```

Comment:
Nitpick: look at various other condition branches in the code base, we typically do not use this code style (per-line conditional check).